### ISSUE TITLE
Offer a filter function in the translate interaction options

### ISSUE THREAD

**Issue by agpixdev** [openlayers/openlayers#9625]
The Translate interaction options currently does not allow to filter which features are to be part of a translate interaction, in a manner similar to what is available for the Select interaction. It would be nice to have the same capability in the Translate interaction.

I can submit a PR for this change, if there is interest.

**Comment by fredj** [comment]
The way to filter the features is to use a Select interaction in combination with the Translate interaction and do the filtering in the Select.

See https://codesandbox.io/s/translatefeatures-xrwsp

In this example, only Brazil and France can be translated because a filter function was added to the Select interaction:
```js
var select = new Select({
  filter: function(feature) {
    var name = feature.get("name");
    return name === "Brazil" || name === "France";
  }
});
```

**Comment by agpixdev** [comment]
Hello Frederic,

Thanks for your comment !

What I'm trying to avoid in this case is the need for the Select interaction as I'd like to be able to translate a feature without first having to click on it for selection, i.e.:

Expected behavior:

1. Move pointer to feature
2. Click on feature (without releasing mouse button) and drag it to new position

Instead of:

1. Move pointer to feature
2. Click on feature, releasing mouse button
3. Click again on feature & drag it to new position

Having the filter function option in the translate interaction would avoid the need to use the Select interaction and the extra click (unless there is another way of doing that that I'm missing).

In the app I'm working on, each feature has flags that indicate if the feature is editable, draggable (a la Google Maps) and the user can start an interaction by simply moving the pointer to the feature.

I've already implemented the change to do this (a few lines of code added) hence the reason I'm proposing this for a pull-request.

Thanks again Frederic !

**Comment by fredj** [comment]
> I've already implemented the change to do this (a few lines of code added) hence the reason I'm proposing this for a pull-request.

yes, please open a pull request

**Comment by fredj** [comment]
fixes with #9638
